src/test/java/cat/udl/eps/softarch/demo/steps/UpdateRequestStepDefs.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    private Request getRequestByParams(String name, int price, String description, String username) {
+        BigDecimal requestPrice = new BigDecimal(price);
+        User requester = getUser(username);
+
+        List<Request> requestList = requestRepository.findByNameAndPriceAndDescriptionAndRequester(name, requestPrice, description, requester);
+        if (requestList.isEmpty()) {
+            throw new NotFoundException();
+        }
+        return requestList.get(0);
+
+    }
+
+    @When("I modify a request with name {string}, price {int}, description {string} by {string} with new price {int}")
+    public void iModifyARequestWithNamePriceDescriptionByWithNewPrice(String name, int price, String description, String username, int newPrice) {
+        try {
+            Long requestId = getRequestByParams(name, price, description, username).getId();
+            System.out.println("AYUDAAAAA " + requestId);
+            stepDefs.result = stepDefs.mockMvc.perform(
+                    patch("/requests/{id}", requestId)
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content((new JSONObject().put("price", new BigDecimal(newPrice))).toString())
+                            .accept(MediaType.APPLICATION_JSON)
+                            .with(AuthenticationStepDefs.authenticate())
+            ).andDo(print());
+        } catch (Exception Nf) {
+            e = Nf;
+        }
+    }
+
+    @Then("The request to modify is not found")
+    public void theRequestToModifyIsNotFound() {
+        Assertions.assertEquals(NotFoundException.class, e.getClass());
+    }
+
+    @When("I modify my own created requests with price {int} \\(put)")
+    public void iModifyMyOwnCreatedRequestsWithPricePut(int newPrice) throws Exception {
+        String currentUser = getCurrentUsername();
+        List<Request> myRequests = requestRepository.findByRequester(getUser(currentUser));
+        Request modifiedRequest = myRequests.get(0);
+        Long requestId = modifiedRequest.getId();
+
+        //System.out.println(modifiedRequest);
+
+        stepDefs.result = stepDefs.mockMvc.perform(
+                put("/requests/{id}", requestId)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content((new JSONObject().put("price", new BigDecimal(newPrice))).toString())
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate())
+        ).andDo(print());
+    }
+
+    @When("I modify {string}'s requests with price {int} \\(put)")
+    public void iModifySRequestsWithPricePut(String othersUsername, int newPrice) throws Exception {
+        List<Request> othersRequests = requestRepository.findByRequester(getUser(othersUsername));
+        Request modifiedRequest = othersRequests.get(0);
+        Long requestId = modifiedRequest.getId();
+
+        //System.out.println(modifiedRequest);
```

Comment:
Please, remove it. If you need to know this value during development, better to use a logger or a breakpoint using the debugger.